### ISSUE TITLE
Tweaking parameters for production -- image size

### ISSUE THREAD

**Issue by twshimwell** [mhardcastle/ddf-pipeline#22]
First should see how imaging time seems to depend on the size. If substantial improvements could revisit subtracting sources and imaging a smaller region than the current default 20,000x20,000 pixels.

**Comment by mhardcastle** [comment]
Only works well (for bright sources in the outer part) if we do some self-cal on the sources to  be subtracted first, no? So the speedup would be in the final selfcal/imaging round.

**Comment by wllwen007** [comment]
Just to note that I was thinking about this myself as I had one case of a bright source just out of the 20k x 20k image from my GAMA fields.

**Comment by cyriltasse** [comment]
I'm branching from pruning. I'm doing 

- DI-SSD
- Cluster in NDir
- kMS in NDir
- DDF Predict+Substract in outerRectangle P only (or AP enabling GlobalNorm, not sure) -> get Predict col for the full mslist
- Continue as before 

It's not as good as what we're currently doing of course but should still produce good results. And it should reduce the computing time by a solid factor of 4.

**Comment by mhardcastle** [comment]
Branch from master please... but otherwise great, look forward to seeing what we get.

**Comment by mhardcastle** [comment]
And perhaps if we are doing this would be worth doing DI and initial selfcal in a larger area with slightly larger pixels to get everything down to the first null? Then back to 10000 x 10000 at 1.5 arcsec.

**Comment by twshimwell** [comment]
Yeh that seems like a good plan. In preparation for factor we image close to 30degx30deg to remove all the sources outside the field (our size with 20000 1.5arcsec pixels is 8.3deg)

**Comment by mhardcastle** [comment]
Not sure we could get decent killms solutions on the long baselines going that far out unless we made a gigantic image. I was thinking maybe going up to 2.5 or 3-arcsec pixels still at 20000x20000.

**Comment by cyriltasse** [comment]
Yeah good idea. We just have to be careful since we use the degridder for the predict, the pixel cannot be too big to compute a predict for a long baseline... This will be up to the user anyway.

**Comment by cyriltasse** [comment]
ok, you can have a try on branch SubSquare.

Just set fact_reduce_field=2 for example in the [image] section in the parset.

also pull on DDFacet (still AutoMasking_SNR branch)

**Comment by twshimwell** [comment]
Hey Cyril. Did you push the SubSquare?

**Comment by cyriltasse** [comment]
yes I think

**Comment by cyriltasse** [comment]
Ha ok, I did not push correctly... now it's done

**Comment by twshimwell** [comment]
Great. Got it now. Is it ok to reuse the image_dirin results from previously? Or do the DDFacet changes prevent that?

**Comment by cyriltasse** [comment]
no, you'll have to rerun from start, removing 

- all killMS* in the MSs
- all *.ddfcache
- all image*

**Comment by mhardcastle** [comment]
I think bootstrap won't work in this version because it runs on the original colname. I'd suggest disabling it for the first tests.

**Comment by cyriltasse** [comment]
It came to me yesterday that we can in principle also compress the data by a factor of 4 once we have lowered the field of view. I'll add this as an addictional option...

**Comment by mhardcastle** [comment]
Yes sure -- we would want to do this for the full speedup.

Could we safely compress _before_ we lower the fov? i.e. we let things be smeared and trust ddf/killms to handle it?

**Comment by cyriltasse** [comment]
Yeah, I was thinking the same actually... we could definitly have a try
